cmd/ftl-controller/main.go

```diff
@@ -53,7 +54,9 @@ func main() {
 	kctx.FatalIfErrorf(err, "failed to initialize observability")
 
 	// The FTL controller currently only supports DB as a configuration provider/resolver.
-	conn, err := sql.Open("pgx", cli.ControllerConfig.DSN)
+	conn, err := otelsql.Open("pgx", cli.ControllerConfig.DSN)
+	kctx.FatalIfErrorf(err)
+	err = otelsql.RegisterDBStatsMetrics(conn, otelsql.WithAttributes(semconv.DBSystemPostgreSQL))
```

Comment:
Can we create a helper function that opens + instruments in one, and use that function in all these locations?

Reply:
great idea - done